utils/config.go

```diff
@@ -221,7 +221,7 @@ func InitializeConfigWatch() {
 						if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
 							l4g.Info(fmt.Sprintf("Config file watcher detected a change reloading %v", CfgFileName))
 
-							if configReadErr := viper.ReadInConfig(); configReadErr == nil {
+							if _, configReadErr := ReadConfigFile(CfgFileName, true); configReadErr == nil {
 								LoadGlobalConfig(CfgFileName)
```

Comment:
On my list of things to refactor really soon. This isn't good error handling at all. It's racey and if an error occurs in the call to `LoadGlobalConfig`, `os.Exit` gets called and the entire server shuts down.